job-simulation.rb

```diff
@@ -1,4 +1,45 @@
-require './Stack.rb'
-require './Queue.rb'
 
+require_relative './Stack.rb'
+require_relative './Queue.rb'
 
+@waiting = Queue.new
+@workers = Stack.new
+
+SPOTS = 6
+
+def self.wait_list(worker_id)
+  # Adds individual workers to the waiting list (workers that are new to the process and have not been hired befored)
+  @waiting.enqueue(worker_id)
+end
+
+def self.fill_spots
+  # Hire people to fill all spots available
+  SPOTS.times do
+    hired = @waiting.dequeue
+    @workers.push(hired)
+  end
+end
+
+
+def self.fire_hire(number)
+  # Validate that number is between 1 and 6 (a roll of dice)
+  unless number > 0 && number < 7
+    raise ArgumentError.new("Only numbers between 1 and 6 are allowed")
+  end
```

Comment:
Oh! I didn't know how to do this in Ruby so I did it a sub-optimal way in my example solution.